spark/src/main/scala/org/apache/spark/sql/delta/OptimisticTransaction.scala

```diff
@@ -1848,6 +1848,37 @@ trait OptimisticTransactionImpl extends TransactionalWrite
     val info = currentTransactionInfo.commitInfo
       .map(_.copy(readVersion = None, isolationLevel = None)).orNull
     setNeedsCheckpoint(attemptVersion, postCommitSnapshot)
+
+    val numFilesTotal =
+      if (spark.sessionState.conf.getConf(DeltaSQLConf.DELTA_COLLECT_COMMIT_STATS)) {
+        // Forces snapshot state reconstruction
+        postCommitSnapshot.numOfFiles
+      } else -1L
+
+    val sizeInBytesTotal =
+      if (spark.sessionState.conf.getConf(DeltaSQLConf.DELTA_COLLECT_COMMIT_STATS)) {
+        // Forces snapshot state reconstruction
+        postCommitSnapshot.sizeInBytes
+      } else -1L
+
+    val protocol =
+      if (spark.sessionState.conf.getConf(DeltaSQLConf.DELTA_COLLECT_COMMIT_STATS)) {
+        // Forces protocolAndMetadata reconstruction
```

Comment:
I dug into it, and you were correct.  By the time we get to this line, P&M is already done.  It gets triggered a few lines earlier [here](https://github.com/delta-io/delta/blob/f0b0878c88294f79562c8d86efbeb8cc94c79270/spark/src/main/scala/org/apache/spark/sql/delta/OptimisticTransaction.scala#L1808-L1813).  I will change it so that protocol is always logged.

> Similarly, under what circumstances might the future at L1872 still be in flight?

The future I'm referring to is [this one](https://github.com/delta-io/delta/blob/f0b0878c88294f79562c8d86efbeb8cc94c79270/spark/src/main/scala/org/apache/spark/sql/delta/CheckpointProvider.scala#L122-L125).  It is relevant if v2 checkpoints are enabled.

To be honest, I have not seen a situation where the Future was still in flight. I _think_ it is technically possible if the optimistic transaction is slower than the future, but I admit that is unlikely.  Furthermore, this is only relevant immediately after a checkpoint.

So I'm going to remove this change too; I will go back to logging the checkpointSizeInBytes every time.  If I ever notice this being a problem then I will revisit it.